packages/axelarnet-gateway-api/src/msg.rs

```diff
@@ -0,0 +1,39 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::HexBinary;
+use router_api::{Address, ChainName, CrossChainId, Message};
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    // Permissionless
+    /// Initiate a cross-chain contract call from Axelarnet to another chain.
+    /// The message will be routed to the destination chain's gateway via the router.
+    CallContract {
+        destination_chain: ChainName,
+        destination_address: Address,
+        payload: HexBinary,
+    },
+
+    // Permissioned: Can only be called by the router
+    /// Receive messages from the router and mark them as approved.
+    RouteMessages(Vec<Message>),
+
+    // Permissioned: Can be called by the address receiving the message
+    /// Validate if the message was received for the caller contract and mark as executed. The receiving contract should call this before executing a message.
```

Comment:
```suggestion
    /// Validate if the message was received for the caller contract and mark as executed.
```

I think it's more likely the contract will actually just add this message as a submessage, which will revert the whole execution on error.